### ISSUE TITLE
Add a Ability to Go back to previous Timerange

### ISSUE THREAD

**Issue by petethepig** [grafana/pyroscope#38]
Multiple people either explicitly mentioned this or had a hard time figuring out how to zoom out. I think we should just have a zoom out button.

Default View on page load:
![image](https://user-images.githubusercontent.com/23323466/105544446-dc4d0780-5caf-11eb-8be2-326f70ac82a7.png)

"Zoomed view" when a yellow bar in the timeline is selected:
![image](https://user-images.githubusercontent.com/23323466/105544511-ec64e700-5caf-11eb-9a9f-87f3d4f0b069.png)

We need a button that allows you to go back to the initial default view. Since URL params change this may be addressable using those as well.

**Comment by LouisInFlow** [comment]
There's a reset view button that appears when zoomed in and disappears when zoomed out, but its not very obvious.

<img width="2992" alt="Screen Shot 2021-01-22 at 11 32 07 PM" src="https://user-images.githubusercontent.com/73438887/105572129-1d720580-5d0a-11eb-91c4-fb77b320ff84.png">

@Rperry2174 What do you think about making the button disabled instead of disappearing. Also a more prominent color?

**Comment by Rperry2174** [comment]
@LouisInFlow sorry for the confusion. Basically we need a "go back to previous time range" button for _the timeline_.

So when you first land on the page the timeline will default to the last hour. So your time range will be `00:00` to `01:00` looking at full 1 hour timespan.

If you drag and drop on the timeline, that is what this issue is referring to. When you click the timeline at `00:30` and drag your mouse over to `00:35`, for example. To look closely at a 5 minute time span.

Someone might do this if they want to inspect what was going on for a particular 5 minutes. Once that person is done looking at those 5 minutes , they probably want to go back to the view where they were looking at the last hour (or whatever timespan preceded their inspection of those 5 minutes).

When @petethepig says "zoom out" this is what he is referring to. It's more of a "go back to previous time range" that we're looking for.

~~~~~

Now this is not to be confused with the "zoom" functionality of the flamegraph itself where you are correct and we have a separate issue addressing the visibility of the "Reset View" button here (#39) 😅  we should use clearer terminology

**Comment by petethepig** [comment]
Got some interesting ideas from Otto:
```
And on the range picker: something that would be awesome is that by selecting a range with
* left button = zooming in
* right button = zooming out
Or
* selecting a range by dragging to the right = zooming in
* selecting a range by dragging to the left = zooming out

or when you double click on the range that it zooms out
```
I think the double click thing makes the most sense — we could have left double click be a zoom out or we could have left double click be a zoom in and right double click be a zoom out action — kinda like in google maps.

**Comment by Rperry2174** [comment]
Thats interesting. I like the advanced functionality, but I think doing something that's basic and visual at least in addition to the double click functionality is better better. 

For example... TIL that double right click zooms out on Google maps. I always just use the plus to zoom in and the minus to zoom out by default and occasionally us double-left-click.

For us, many dont realize that they can even interact with the timeline directly. If we had visual buttons on it, then it would be clear that you can zoom in or out

**Comment by Rperry2174** [comment]
As a starting point I think we should add a button with a dropdown arrow something like this:
![image](https://user-images.githubusercontent.com/23323466/127035511-fdc9abb3-6d1a-4720-8f02-d9b87cdddcf9.png)

Then when a user clicks on the dropdown arrow they should be able to see a dropdown with the most recently selected time ranges:
![image](https://user-images.githubusercontent.com/23323466/127035666-cd340028-6b64-4cf7-83a1-a73bd635d70a.png)

We may need to play around with the UI aspect of this so that the buttons don't get confusing (or add tooltips), but once we have the functionality we can figure out what the best UI/UX for it is
